### DISCUSSION TITLE
STATUS_DLL_NOT_FOUND error (Exit code 0xC0000135)

### DISCUSSION THREAD

**Discussion by ldorau** [community/community#25091] [Actions]
<p>Hi,</p>
<p>I have got the STATUS_DLL_NOT_FOUND error (Exit code 0xC0000135).</p>
<p>I have run <code>dumpbin.exe" /DEPENDENTS</code> on the binary and listed dependent DLLs and checked that all of them are in the PATH, so what can be the cause of this error? What DLL can be missing?</p>
<p>The whole log can be found in the failed step “Test” here:</p>
<p><a href="https://github.com/ldorau/libpmemobj-cpp/commit/17c89bff6dffc1b040e8495a434d636c80d5a643/checks?check_suite_id=341850102" rel="nofollow noopener">https://github.com/ldorau/libpmemobj-cpp/commit/17c89bff6dffc1b040e8495a434d636c80d5a643/checks?check_suite_id=341850102</a></p>

**Comment by Yanjingzhu** [comment]
<p>Can you check whether your test <strong>ex-queue_0_none</strong> is correct? Could <strong>ex-queue_0_none</strong> test be executed successfully in your local machine?</p>

**Comment by ldorau** [comment]
<blockquote>
<p><a class="mention" href="/yanjingzhu">@yanjingzhu</a> wrote:</p>
</blockquote>
<blockquote>
<p>Can you check whether your test <strong>ex-queue_0_none</strong> is correct? Could <strong>ex-queue_0_none</strong> test be executed successfully in your local machine?</p>
</blockquote>
<p>Yes, it is correct.</p>
<p>As you can see in: </p>
<p><a href="https://github.com/ldorau/libpmemobj-cpp/commit/b52623eef70cf104c11b8b64241b132752823fba/checks?check_suite_id=348762827" rel="nofollow noopener">https://github.com/ldorau/libpmemobj-cpp/commit/b52623eef70cf104c11b8b64241b132752823fba/checks?check_suite_id=348762827</a></p>
<p>The command:</p>
<p>&gt; D:/a/libpmemobj-cpp/libpmemobj-cpp/build/tests/Debug/ex-queue.exe ./testfile push 1</p>
<ul>
<li>
<p><strong>succeeds</strong> when it is called directly in the shell (lines <span class="hashtag">#82-83</span> in the log) and</p>
</li>
<li>
<p><strong>fails</strong> when it is called by ctest (lines <span class="hashtag">#147</span> <span class="hashtag">#154</span> in the log).</p>
</li>
</ul>
<p>So it can be a problem with CMake or ctest.</p>

**Comment by Yanjingzhu** [comment]
<p>Can you use cTest to run the test in your local machine? Would you like to <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/managing-a-workflow-run#enabling-step-debug-logging" rel="nofollow noopener">enable debug logging</a> ,run a new workflow run and share us the workflow run link here?  </p>

**Comment by ldorau** [comment]
<blockquote>
<p><a class="mention" href="/yanjingzhu">@yanjingzhu</a> wrote:</p>
</blockquote>
<blockquote>
<p>Can you use cTest to run the test in your local machine? Would you like to <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/managing-a-workflow-run#enabling-step-debug-logging" rel="nofollow noopener">enable debug logging</a> ,run a new workflow run and share us the workflow run link here?  </p>
</blockquote>
<p>I run these tests with CTest on my local (but Linux) machine and it works.</p>
<p>The same tests run with CTest on Appveyor (on Windows OS) also work:</p>
<p><a href="https://ci.appveyor.com/project/pmem/libpmemobj-cpp/builds/29466753" rel="nofollow noopener">https://ci.appveyor.com/project/pmem/libpmemobj-cpp/builds/29466753</a></p>
<p><a href="https://github.com/pmem/libpmemobj-cpp/blob/7191ebea48f9d1bde6abc73ab08b2604c48d4f96/appveyor.yml" rel="nofollow noopener">https://github.com/pmem/libpmemobj-cpp/blob/7191ebea48f9d1bde6abc73ab08b2604c48d4f96/appveyor.yml</a></p>
<p>I have <strong>enabled debug logging</strong> :</p>
<p><a href="https://github.com/ldorau/libpmemobj-cpp/commit/e4601cfb1bba330267c4bef3cbff2ff25da44047/checks?check_suite_id=353327785" rel="nofollow noopener">https://github.com/ldorau/libpmemobj-cpp/commit/e4601cfb1bba330267c4bef3cbff2ff25da44047/checks?check_suite_id=353327785</a></p>
<p>The code can be found here:</p>
<p><a href="https://github.com/ldorau/libpmemobj-cpp/tree/windows" rel="nofollow noopener">https://github.com/ldorau/libpmemobj-cpp/tree/windows</a></p>
<p>Thanks!</p>

**Comment by ldorau** [comment, marked as answer]
<p>The root cause and the solution of the issue:</p>
<p>I forgot to set the architecture for CMake via the ‘-A’ option:</p>
<pre><code>-A "x64"
</code></pre>
<p>e.g.:</p>
<pre><code>cmake . -G "Visual Studio 16 2019" -A "x64" ...
</code></pre>
